### PR TITLE
Samus Returns - Initial prep for patcher support

### PR DESCRIPTION
**Pull request by dyceron**

### CODE REVIEW

File: randovania/games/samus_returns/gui/__init__.py [hunk 1]
```diff
@@ -8,4 +8,4 @@
     MSRCosmeticPatchesDialog,
     MSRGameExportDialog,
     preset_tabs,
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/gui/__init__.py:10]: You named that `msr_preset_tabs` in the meantime

File: randovania/games/samus_returns/exporter/patch_data_factory.py [hunk 2]
```diff
@@ -1,10 +1,146 @@
+import os
+
+from randovania.exporter import pickup_exporter
 from randovania.exporter.patch_data_factory import BasePatchDataFactory
+from randovania.exporter.pickup_exporter import ExportedPickupDetails
+from randovania.game_description.assignment import PickupTarget
+from randovania.game_description.resources.item_resource_info import ItemResourceInfo
+from randovania.game_description.resources.pickup_entry import ConditionalResources
+from randovania.game_description.resources.resource_info import ResourceCollection
+from randovania.game_description.world.node import Node
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/exporter/patch_data_factory.py:10]: All these imports of `randovania.game_description.world......` moved to `randovania.game_description.db....` 
Also`world` was renamed to `region`.
Please rename also the variables

File: randovania/games/samus_returns/exporter/patch_data_factory.py [hunk 3]
```diff
@@ -1,10 +1,146 @@
...
+            try:
+                result[get_item_id_for_item(resource)] = quantity
+            except KeyError:
+                print(f"Skipping {resource} for starting inventory: no item id")
+                continue
+        return result
+
+    def _node_for(self, identifier: AreaIdentifier | NodeIdentifier) -> Node:
+        if isinstance(identifier, NodeIdentifier):
+            return self.game.world_list.node_by_identifier(identifier)
```
Comment (ThanatosGit) [inline randovania/games/samus_returns/exporter/patch_data_factory.py:81]: e.g. this is now `region_list` at probably many more usages of world are in here

**Review by ThanatosGit — CHANGES_REQUESTED** [review]
This is just for passing tests

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
Please test `MSRPatchDataFactory.create_data`.

Look at test_dread_patch_data_factory and test_create_patch_data there for a sample code.
You'll have to generate a RDVGame, add to test files folder and then also add the expected patch_data.json

**Pull request closed without merging**